drivers/hyperv/hyperv.go

```diff
@@ -428,6 +452,10 @@ func (d *Driver) Kill() error {
 	return nil
 }
 
+func isIPv4(address string) bool {
```

Comment:
What we do in Minishift is as follows: https://github.com/minishift/minishift/blob/0efa5d527d77a7651eac75a4861097ab4a203f43/cmd/minishift/cmd/start_preflight.go#L485-L491

```
func isIPv4(address string) bool {
   return net.ParseIP(ip).To4() != nil
}
```

should be enough and relies on the framework instead of counting delimiters.

`127.1` is a valid IPv4 address... which expands to `127.0.0.1`. Does your code work with this?

Reply:
I thought about using `To4()` for checking.
However, IPv4 address converted to IPv6 format `0:0:0:0:0:ffff:d1ad:35a7` could be considerred as IPv4 address since `To4` can take both v4 and v6 address.
```
	if len(ip) == IPv6len &&
		isZeros(ip[0:10]) &&
		ip[10] == 0xff &&
		ip[11] == 0xff {
		return ip[12:16]
	}
```
I'm not sure which format could be accepted by minikube (either converting it back to IPv4 or still using this IPv6 format). Therefore, I used delimiter counting instead of `To4()` here to avoid this potential issue.

Reply:
does Minikube function with any IPv6 address?
We have seen issues with the SSH clients not accepting the address (espcially on Windows).

Reply:
But with `IPv4` the correct result is returned anyway `return ip[12:16]` so shouldn't that then we used instead on the switch case for IPv4 protocol? We do not mind if IPv6 is returned from the networkadapter, as the network stack should handle this for us anyway when this is instead resolved with the relevant IPv4 format.

Reply:
I was not sure which format v4 or v6 should be returned in this case as the function asks for a string.
It's not clear that if IPv4 address in v6 format could be accepted in Windows or simply just return a value in IPv4.
Let me do some experiment on this.

Reply:
It doesn't seem IPv4 address in IPv6 format functions well on Windows.
Thus I would like to keep the code to count on ":" instead of `To4` in which case a closer match for IPv4 format can be done and only IPv4 format address could be returned.